pkg/service-manager/handlers.go

```diff
@@ -0,0 +1,281 @@
+// Copyright (c) 2017 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+///////////////////////////////////////////////////////////////////////
+
+package servicemanager
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/vmware/dispatch/pkg/controller"
+	"github.com/vmware/dispatch/pkg/utils"
+
+	"github.com/go-openapi/runtime/middleware"
+	"github.com/go-openapi/swag"
+	log "github.com/sirupsen/logrus"
+
+	entitystore "github.com/vmware/dispatch/pkg/entity-store"
+	"github.com/vmware/dispatch/pkg/service-manager/entities"
+	"github.com/vmware/dispatch/pkg/service-manager/flags"
+	"github.com/vmware/dispatch/pkg/service-manager/gen/models"
+	"github.com/vmware/dispatch/pkg/service-manager/gen/restapi/operations"
+	serviceclass "github.com/vmware/dispatch/pkg/service-manager/gen/restapi/operations/service_class"
+	serviceinstance "github.com/vmware/dispatch/pkg/service-manager/gen/restapi/operations/service_instance"
+	"github.com/vmware/dispatch/pkg/trace"
+)
+
+// Handlers encapsulates the service manager handlers
+type Handlers struct {
+	Store   entitystore.EntityStore
+	Watcher controller.Watcher
+}
+
+// NewHandlers is the constructor for the Handlers type
+func NewHandlers(watcher controller.Watcher, store entitystore.EntityStore) *Handlers {
+	defer trace.Trace("")()
+	return &Handlers{
+		Store:   store,
+		Watcher: watcher,
+	}
+}
+
+// ConfigureHandlers registers the service manager handlers to the API
+func (h *Handlers) ConfigureHandlers(api middleware.RoutableAPI) {
+	defer trace.Trace("")()
+	a, ok := api.(*operations.ServiceManagerAPI)
+	if !ok {
+		panic("Cannot configure api")
+	}
+
+	entities.InitializeStatusMap()
+
+	a.CookieAuth = func(token string) (interface{}, error) {
+
+		// TODO: be able to retrieve user information from the cookie
+		// currently just return the cookie
+		log.Printf("cookie auth: %s\n", token)
+		return token, nil
+	}
+	a.ServiceClassGetServiceClassByNameHandler = serviceclass.GetServiceClassByNameHandlerFunc(h.getServiceClassByName)
+	a.ServiceClassGetServiceClassesHandler = serviceclass.GetServiceClassesHandlerFunc(h.getServiceClasses)
+
+	a.ServiceInstanceAddServiceInstanceHandler = serviceinstance.AddServiceInstanceHandlerFunc(h.addServiceInstance)
+	a.ServiceInstanceGetServiceInstanceByNameHandler = serviceinstance.GetServiceInstanceByNameHandlerFunc(h.getServiceInstanceByName)
+	a.ServiceInstanceGetServiceInstancesHandler = serviceinstance.GetServiceInstancesHandlerFunc(h.getServiceInstances)
+	a.ServiceInstanceDeleteServiceInstanceByNameHandler = serviceinstance.DeleteServiceInstanceByNameHandlerFunc(h.deleteServiceInstanceByName)
+}
+
+func (h *Handlers) getServiceClassByName(params serviceclass.GetServiceClassByNameParams, principal interface{}) middleware.Responder {
+	defer trace.Trace("")()
+	e := entities.ServiceClass{}
+	err := h.Store.Get(flags.ServiceManagerFlags.OrgID, params.ServiceClassName, entitystore.Options{}, &e)
```

Comment:
Why not use the new find method to distinguish between not found and internal error?